src/pal/src/misc/sysinfo.cpp

```diff
@@ -495,3 +495,33 @@ PAL_GetLogicalProcessorCacheSizeFromOS()
 
     return cacheSize;
 }
+
+size_t
+PALAPI
+PAL_GetMemAvailableFromOS()
```

Comment:
This would not work as you intend when running in a docker container. The /proc/meminfo reports the memory in the whole system.

Reply:
@janvorli I've added checking for docker